### PR TITLE
add ut for istioctl/pkg/authz/listener_test.go

### PR DESCRIPTION
**Pull request by hexiaodai**
**Please provide a description of this PR:**

add ut for istioctl/pkg/authz/listener_test.go

I want to contribute a little code.

### CODE REVIEW

**Review by GregHanson — COMMENTED** [review]
we tend to defer tests that that don't add much value - and I think this is close to falling into this category.  I think a more valuable addition would be testing the top-level `Print` method which calls `extractName` from further up in the call stack.  `istioctl/cmd/authz_test.go` is basically an empty file right now.  I'll defer approval to @hanxiaop

**Comment by hanxiaop** [comment]
> we tend to defer tests that that don't add much value - and I think this is close to falling into this category. I think a more valuable addition would be testing the top-level `Print` method which calls `extractName` from further up in the call stack. `istioctl/cmd/authz_test.go` is basically an empty file right now. I'll defer approval to @hanxiaop

I agree with @GregHanson. I think the outer `Print` function is a more proper layer to add tests, or the analyzer, maybe we can add tests to that analyzer like other analyzers.

File: istioctl/pkg/authz/listener_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			name:          "no ns",
+			input:         "policy[some-policy]-rule[1]",
+			expectedName:  "",
+			expectedIndex: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if actualName, actualIndex := extractName(tt.input); actualName != tt.expectedName || actualIndex != tt.expectedIndex {
+				t.Errorf("extractName() got %v %v, want %v %v", actualName, actualIndex, tt.expectedName, tt.expectedIndex)
```
Comment (hanxiaop) [inline istioctl/pkg/authz/listener_test.go:86]: The names are all generated, and I think it doesn't add much value for testing these.

**Pull request closed without merging**